### ISSUE TITLE
TF Swiftformer

### ISSUE THREAD

**Issue by amyeroberts** [huggingface/transformers#22771]
### Model description

Add the TensorFlow port of the SwiftFormer model. See related issue: #22685 

To be done once the SwiftFormer model has been added: #22686 

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

Original repo: https://github.com/amshaker/swiftformer

**Comment by joaocmd** [comment]
Hi! I would like to take on this as my first issue if possible. Is that okay?

**Comment by amyeroberts** [comment]
@joaocmd Of course! Cool issue to start on 🤗 

If you haven't seen it already, there's a detailed guide in the docs on porting a model to TensorFlow: https://huggingface.co/docs/transformers/add_tensorflow_model

It's best to wait until the PyTorch implementation is merged in, which will be at least a day or two away.

**Comment by joaocmd** [comment]
Sounds great @amyeroberts, thank you :)

I'll start looking into it once the PyTorch PR is merged.

**Comment by D-Roberts** [comment]
Any news on this tf model?

**Comment by joaocmd** [comment]
Hi @D-Roberts, I haven't started looking into it as the pytorch version has not yet been merged.

**Comment by shehanmunasinghe** [comment]
Hi @joaocmd , the PyTorch version of SwiftFormer is now merged so you can continue working the TensorFlow version of it.

**Comment by joaocmd** [comment]
Hi @shehanmunasinghe, I'm on it, thanks!

**Comment by sqali** [comment]
Hi @amyeroberts ,

May I proceed to work this issue if it has not been sorted yet?

**Comment by amyeroberts** [comment]
@sqali There is currently an open PR which is actively being worked on by @joaocmd: #23342

**Comment by sqali** [comment]
Hi @amyeroberts ,

Is there any other issue that you are working on in which I can help you with?

**Comment by amyeroberts** [comment]
@sqali Here's a page in the docs all about different ways to contribute and how to find issues to work on: https://huggingface.co/docs/transformers/main/contributing. I'd suggest looking at issues marked with the 'Good First Issues' tag and finding one which no-one is currently working on: https://github.com/huggingface/transformers/labels/Good%20First%20Issue
